bin/fix_permissions

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```

Comment:
Would you mind converting this to python (at least, the same but with plumbum)? We have a goal here of getting rid of as much bash as possible (see #7).

That way we get more readability (the code is a bit cumbersome.. not that it could be better, bash is this way...), and code formatting, and all that stuff for free.

Reply:
I'm not sure plumbum would make this more readable but i can try.

Reply:
One of the advantages here is that it could drop the pipelined commands and the usage of `xargs` and instead use Python to read the output from one command, parse it, and pass it to the other, making it a lot more readable.

Reply:
well the point of this merge request is to make the build of scaffolding much faster (and therefore the tests). i'm pretty sure inventing our own xargs and find in python will not run as fast as just using the tools other people already worked hard on optimizing. in my experience custom code never runs as fast as the core linux utils. 

I'll still try to implement this. But i also then need to verify the speed again (and i already worked a few weeks for this on our systems), so please be patient with me.

Reply:
Well, of course, if the lost in performance is big, it might not be worth it, but the idea @Yajo was mentioning was to use Plumbum to call those Linux tools (`find`, `chown`) and just use Python to control the flow and parse the results, that shouldn’t introduce a big overhead... 

> I'll still try to implement this. But i also then need to verify the speed again (and i already worked a few weeks for this on our systems), so please be patient with me.

Don’t worry, the contribution is very appreciated 🙂

Reply:
> i'm pretty sure inventing our own xargs and find in python will not run as fast as just using the tools other people already worked hard on optimizing

You can still do the pipelining in plumbum and use xargs too.

At the end of the day, `bash` spawns a new subprocess for every command. Replacing that with `python` shouldn't impact negatively in performance (at least I never noticed anything significant). However, an easier to read syntax makes code easier to reason about. And we have linters, formatters, and (more important) more human experience on that language, so it helps in the long run.

Reply:
Well as plumbum is python i wonder if some magic env variables for python have an effect on it. (E.g. -u or PYTHONUNBUFFERED to get stdout and stderr synced in python also seems to affect file io and i would expect also affecting pipelining in plumbum, but that's all on the test plan)

Reply:
Just one more detail. I just realized we have the `build.d/800-permissions` file which was meant to fix permissions.

Probably it would make more sense to move this code there.

Reply:
well i was moving it to a seperate bin file because i wanted consistent permissions across all steps (to prevent changing permissions to often) and as bin is present all the way (build + run time) in the container but build.d sounded like someday it could be present just during build time. so i didn't want bin/autoaggregate invoking something in build.d and instead invoke bin/fix_permissions from bin/autoaggregate and build.d/800-fix_permissions

Reply:
Also it might just be a java habbit in me to not cross invoke package
s (build.d -> bin -> build.d -> bin ...) but instead have a clean one directional dependency (build.d -> bin but never bin -> build.d)

If that's not an issue for you i can jump over my own shadow and just invoke build.d/800-fix_permissions in bin/autoaggregate. (and it might not be needed after all)